### PR TITLE
Add option to only show custom modules

### PR DESCRIPTION
**Pull request by norgeindian**
Usually it is not possible / recommended to adapt third party modules. They should be updated by the vendor directly.
So I think it would be great to have the possibility to only show custom modules.
This is why I added an option to only show modules with specific namespaces.

### CODE REVIEW

File: src/Ampersand/PatchHelper/Command/AnalyseCommand.php [hunk 1]
```diff
@@ -56,7 +58,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
                 $output->writeln("<info>Validating $file</info>", OutputInterface::VERBOSITY_VERBOSE);
 
-                foreach ($patchOverrideValidator->validate()->getErrors() as $errorType => $errors) {
+                $vendorNamespaces = [];
+                if ($input->getOption('vendor-namespaces')) {
+                    $vendorNamespaces = explode(',', $input->getOption('vendor-namespaces'));
```
Comment (sprankhub) [inline src/Ampersand/PatchHelper/Command/AnalyseCommand.php:63]: I recommend trimming the namespaces, so that `"Namespace1, Namespace2"` works :)

Comment (norgeindian) [reply src/Ampersand/PatchHelper/Command/AnalyseCommand.php]: Good point, thanks. Did that now.

**Comment by convenient** [comment]
Hello

Just so I understand the use case here. If I know that the only customisations in my project are `Ampersand_X` modules, this would allow me to run this and filter out for just those modules?

Also 4 pull requests in a few hours, is there a hackathon or something ongoing?

**Comment by norgeindian** [comment]
@convenient, exactly that is the idea. You can also add several vendors and filter for example for `Ampersand,Client`. At least that is the way we usually name our modules, so either after our company or after the client.
Indeed, we currently have an Online Hackathon going on and building a module, which solves as much as possible during updates was one of the projects.
Fortunately we found this great repo, which already helps a lot, so we decided to add our ideas here directly.
Hope you like them...

**Comment by convenient** [comment]
Cool.

Currently the tests use the same command for all runs

https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/blob/master/dev/phpunit/functional/FunctionalTests.php#L10

I don't see why we couldn't use a specific command for a specific test to ensure that `vendor-namespaces` works as we expect? I had actually found this kind of problem before when writing these tests, I only wanted to assert on `ampersand` namespace and you can see I've hackily filtered the test data here to make it easier to assert 

https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/blob/master/dev/phpunit/functional/FunctionalTests.php#L28-L34

I'm aware this is a pretty low risk change, but having test coverage makes future work and changes much more reliable. Let me know what you think thanks.

**Comment by convenient** [comment]
Hey @norgeindian I've tweaked the test suite to cater for this. It's gone green so great job on this.

**Comment by norgeindian** [comment]
@convenient , thanks a lot for your work on that and thanks for merging.

**Pull request merged**